File: app/src/main/java/com/ty/zbpet/ui/adapter/material/MaterialDoneDetailAdapter.java
package com.ty.zbpet.ui.adapter.material;

import android.content.Context;

import com.ty.zbpet.R;
import com.ty.zbpet.bean.material.MaterialDetailsOut;
import com.zhy.adapter.recyclerview.CommonAdapter;
import com.zhy.adapter.recyclerview.base.ViewHolder;

import java.util.List;

/**
 * @author TY on 2018/11/14.
 * 已办详情
 */
public class MaterialDoneDetailAdapter extends CommonAdapter<MaterialDetailsOut.ListBean> {


    public MaterialDoneDetailAdapter(Context context, int layoutId, List datas) {
        super(context, layoutId, datas);
    }

    @Override
    protected void convert(ViewHolder holder, MaterialDetailsOut.ListBean listBean, int position) {

        holder.setText(R.id.tv_name, listBean.getMaterialName())
                .setText(R.id.tv_num, listBean.getNumber())
                .setText(R.id.tv_bulk, "含量：" + listBean.getConcentration() + "%")
                .setText(R.id.tv_zkg, listBean.getZKG())
                .setText(R.id.tv_count_number, listBean.getNumber())
                .setText(R.id.tv_sap, listBean.getSapMaterialBatchNo())
                .setText(R.id.tv_code, listBean.getPositionNo());

    }

}

File: app/src/main/java/com/ty/zbpet/ui/adapter/material/MaterialDoneDetailAdapter.kt
package com.ty.zbpet.ui.adapter.material

import android.content.Context

import com.ty.zbpet.R
import com.ty.zbpet.bean.material.MaterialDetails
import com.zhy.adapter.recyclerview.CommonAdapter
import com.zhy.adapter.recyclerview.base.ViewHolder

/**
 * @author TY on 2018/11/14.
 * 已办详情
 */
class MaterialDoneDetailAdapter(context: Context, private var voucherNo: String?, layoutId: Int, datas: List<MaterialDetails.ListBean>)
    : CommonAdapter<MaterialDetails.ListBean>(context, layoutId, datas) {

    override fun convert(holder: ViewHolder, listBean: MaterialDetails.ListBean, position: Int) {

        holder.setText(R.id.tv_voucher_no, "物料凭证号：$voucherNo")
                .setText(R.id.tv_document_no, "采购子单号：${listBean.sapOrderNo}")
                .setText(R.id.tv_material_name, "原辅料名称：${listBean.materialName}")
                .setText(R.id.tv_number, "入库数量：${listBean.number}")
                .setText(R.id.tv_solubility, "含量：${listBean.concentration}")
                .setText(R.id.tv_batch_number, "SAP 批次号：${listBean.sapBatchNo}")

    }

}
